java-manta-client-unshaded/src/main/java/com/joyent/manta/client/crypto/EncryptionContext.java

```diff
@@ -41,15 +41,23 @@
      */
     private final Cipher cipher;
 
+    /**
+     * Flag indicating that we will only use cloneable ciphers for encryption
+     * operations.
+     */
+    private final boolean useCloneableCipher;
+
     /**
      * Creates a new instance of an encryption context.
      *
-     * @param key           secret key to initialize cipher with
+     * @param key secret key to initialize cipher with
      * @param cipherDetails cipher/mode properties object to create cipher object from
+     * @param useCloneableCipher true when only cloneable ciphers can be used
      */
     public EncryptionContext(final SecretKey key,
-                             final SupportedCipherDetails cipherDetails) {
-        this(key, cipherDetails, null);
+                             final SupportedCipherDetails cipherDetails,
+                             final boolean useCloneableCipher) {
```

Comment:
I've read this a few times and I'm unsure if `useCloneableCipher` means a cloneableCipher *must* be used, or *may* be used.

Does "true when only cloneable ciphers can be used" mean that a cloneableCipher can be used with `useCloneableCipher=false` (the opposite of "when only")?

Reply:
I've renamed it to `requireCloneableCipher`. I think that is clearer.